### DISCUSSION TITLE
engine-twig-php: @basalt/twig-renderer and PHP 8.1

### DISCUSSION THREAD

**Discussion by hadl** [pattern-lab/patternlab-node#1474] [Q&A]
Hi,

does anyone know if the https://github.com/knapsack-cloud/twig-renderer package is dead?
There are some important PRs from dependabot - e.g. https://github.com/knapsack-cloud/twig-renderer/pull/98 which would be really needed to make Twig run without a lot of deprecation notices inside the twig package and PHP 8.1 (i haven't tried 8.2 yet).

There is another package https://www.npmjs.com/package/@knapsack/renderer-twig - but i guess they update that only for "their" software (its private).

Would it make sense to move that dependency into the PL namespace and keep that package alive for "us"?
Or do you have a different approach?

-----

For anyone running into the same issue:
You have to set `error_reporting = E_ALL & ~E_DEPRECATED & ~E_USER_DEPRECATED` in your php.ini to suppress them.
But you can't do that only for the twig-renderer and setting it globally maybe won't help with everyday work. 

At least for me using a CMS where you should keep an eye on the deprecations for future updates and disabling the error_reporting setting back and forth is simply annoying 😇

Thanks!
Andi

**Comment by mfranzke** [comment]
@sghoweri and @EvanLovely, could you please answer this question regarding the outlook on https://github.com/knapsack-cloud/twig-renderer development?

**Comment by mfranzke** [comment]
@hadl my interpretation is that as e.g. the package https://www.npmjs.com/package/@knapsack/renderer-twig (as well as others like e.g. https://www.npmjs.com/package/eleventy-plugin-twig-php) are depending on it, I would assume that the `@basalt/twig-renderer` package they are based on would still get updated the sooner the later, especially as `@knapsack/renderer-twig` is still being updated.

**Comment by hadl** [comment]
Just saw that the twig-renderer was updated to v3.0.1 and is using Twig V3 now.
I am currently running Patternlab in a new project with the new version and at a first glance there are no issues.
Will keep an eye on that and then would suggest to update the dependency in PL too.

**Updates:**

After running PL with the new version without any issues i created PR https://github.com/pattern-lab/patternlab-node/pull/1499

**Reply by EvanLovely** [reply]
@hadl Yep, we updated to use Twig 3 in that. Glad you updated Pattern Lab and are able to use it!
